tests/baselines/reference/isolated-declarations/auto-fixed/diff/expandoFunctionExpressionsWithDynamicNames.d.ts.diff

```diff
@@ -0,0 +1,53 @@
+// [[Reason: undefined]] ////
+
+//// [tests/cases/compiler/expandoFunctionExpressionsWithDynamicNames.ts] ////
+
+===================================================================
+--- TSC	declarations
++++ DTE	declarations
+@@ -1,12 +1,34 @@
+ 
+ 
+ //// [expandoFunctionExpressionsWithDynamicNames.d.ts]
+-export declare const expr: {
+-    (): void;
+-    X: number;
+-};
+-export declare const expr2: {
+-    (): void;
+-    X: number;
+-};
+-//# sourceMappingURL=expandoFunctionExpressionsWithDynamicNames.d.ts.map
+\ No newline at end of file
++export declare const expr: invalid;
++export declare const expr2: invalid;
++//# sourceMappingURL=expandoFunctionExpressionsWithDynamicNames.d.ts.map
++
++/// [Errors] ////
++
++expandoFunctionExpressionsWithDynamicNames.ts(5,14): error TS9007: Declaration emit for this file requires type resolution. An explicit type annotation may unblock declaration emit.
++expandoFunctionExpressionsWithDynamicNames.ts(5,14): error TS9009: Assigning properties to functions without declaring them is not supported with --isolatedDeclarations. Add an explicit declaration for the properties assigned to this function.
++expandoFunctionExpressionsWithDynamicNames.ts(8,14): error TS9007: Declaration emit for this file requires type resolution. An explicit type annotation may unblock declaration emit.
++expandoFunctionExpressionsWithDynamicNames.ts(8,14): error TS9009: Assigning properties to functions without declaring them is not supported with --isolatedDeclarations. Add an explicit declaration for the properties assigned to this function.
++
++
++==== expandoFunctionExpressionsWithDynamicNames.ts (4 errors) ====
++    // https://github.com/microsoft/TypeScript/issues/54809
++    
++    const s = "X";
++    
++    export const expr = (): void => {}
++                 ~~~~
++!!! error TS9007: Declaration emit for this file requires type resolution. An explicit type annotation may unblock declaration emit.

```

Comment:
This shouldn't have triggered as it does have type annotation on the RHS. 
1. empty parameter.
2. void as a return type.
Triggering TS9009 should be enough.